pkg/apis/core/types.go

```diff
@@ -2787,6 +2787,22 @@ type Sysctl struct {
 	Value string
 }
 
+// PodFSGroupChangePolicy holds policies that will be used for applying fsGroup to a volume
+// when volume is mounted.
+type PodFSGroupChangePolicy string
+
+const (
+	// OnRootMismatch indicates that volume's ownership and permissions will be changed
+	// only when permission and ownership of root directory does not match with expected
+	// permissions on the volume. This can help shorten the time it takes to change
+	// ownership and permissions of a volume.
+	OnRootMismatch PodFSGroupChangePolicy = "OnRootMismatch"
```

Comment:
The convention for most of the consts in the API is to prefix with some common name (see reclaim policy as an example). What if we name these something like "FSGroupChangeOnRootMismatch", "FSGroupChangeAlways"?

Reply:
This was discussed in https://github.com/kubernetes/enhancements/pull/1478#discussion_r371464115  . One of @thockin's concern were having too many letters in these types. May be we can keep variable name to be `FSGroupChangeOnRootMismatch` but value to be `OnRootMismatch`, so as user do not have to type as much?

Reply:
I mean in the variable name. The string value I'm fine with the way it currently is.

Reply:
fixed